### PR TITLE
Null guard in `ShellFlyoutTemplatedContentRenderer.HeaderContainer.UpdateMinimumHeight`

### LINKED ISSUE #13825: [Android] NullReferenceException in `ShellFlyoutTemplatedContentRenderer.HeaderContainer.UpdateMinimumHeight`
### Description

Report in AppCenter of a NullReferenceException in `ShellFlyoutTemplatedContentRenderer.HeaderContainer.UpdateMinimumHeight`

I haven't seen this crash before 7.0.59, but I can't say it isn't just something rare I haven't yet seen.

Opportunities for nulls are:
* `Context` on line 772, 776
* `PlatformView` on line 792

Assuming the isolated error can be addressed, perhaps line 792 should at a minimum change to `PlatformView?.MinimumHeight != minHeight)` ?

https://github.com/dotnet/maui/blob/ae20455793142ae0034bcfdf8f810ca66aec782a/src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs#L766-L796

### Steps to Reproduce

Unknown

### Link to public reproduction project repository

not yet

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android

### Affected platform versions

Android 10

### Did you find any workaround?

_No response_

### Relevant log output

```shell
Microsoft.Maui.Controls.Platform.Compatibility.ShellFlyoutTemplatedContentRenderer.HeaderContainer.UpdateMinimumHeight()
Microsoft.Maui.Controls.Platform.Compatibility.ShellFlyoutTemplatedContentRenderer.HeaderContainer.OnMeasure(Int32 widthMeasureSpec, Int32 heightMeasureSpec)
Android.Views.View.n_OnMeasure_II(IntPtr , IntPtr , Int32 , Int32 )
Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PPII_V(_JniMarshal_PPII_V , IntPtr , IntPtr , Int32 , Int32 )
```
```

### PR DESCRIPTION
**Pull request by BioTurboNick**
### Description of Change

Regular NullReferenceExceptions here, as described in #13825 
Absent guidance, trying to push this ahead. Hopefully it can be backported to 7.0.

### Issues Fixed

Fixes #13825

### CODE REVIEW

File: src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs [hunk 1]
```diff
@@ -789,7 +789,7 @@ void UpdateMinimumHeight()
 					frameLayoutView.SetMinimumHeight(minHeight);
 				}
 
-				if (PlatformView.MinimumHeight != minHeight)
+				if (PlatformView?.MinimumHeight != minHeight)
```
Comment (PureWeen) [inline src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:792]: Should we check if `PlatformView is not null` here?

If `PlatformView` is null here, I think it'll fall through to line 794 and that will now throw a NRE. 

Is there anything you can tell me about your use of `Flyouts` in your app? or about the `Shell` structure of your app? Do you swap the content out? are you just using FlyoutItems?

Comment (BioTurboNick) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:792]: Oh, duh... I shouldn't PR late at night. That's what I meant to do.

I don't think I'm doing anything special, this is the complete definition for the AppShell XAML:
```
<?xml version="1.0" encoding="UTF-8" ?>
<Shell
    x:Name="root"
    x:Class="HemocytometerSidekick.AppShell"
    xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
    xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
    xmlns:local="clr-namespace:HemocytometerSidekick"
    xmlns:views="clr-namespace:HemocytometerSidekick.Views"
    xmlns:core="clr-namespace:HemocytometerSidekick.Core;assembly=HemocytometerSidekick.Core"
    FlyoutBackgroundColor="{AppThemeBinding Dark={StaticResource DarkBackgroundColor}}">

    <Shell.FlyoutHeader>
        <Grid
            BackgroundColor="{AppThemeBinding Light={StaticResource Primary}, Dark={StaticResource PrimaryMediumDark}}"
            ColumnDefinitions="10,80,*,10"
            RowSpacing="5"
            RowDefinitions="30,80,Auto"
            Padding="0, 0, 0, 10">

            <Image
                Grid.Row="1"
                Grid.Column="1"
                Source="menulogo.png" />

            <Label
                Grid.Column="1"
                Grid.Row="2"
                Grid.ColumnSpan="2"
                TextColor="White"
                FontSize="20"
                Text="{x:Static core:Resources.Strings.AppResources.AppName}" />
            
        </Grid>
    </Shell.FlyoutHeader>

    <ShellContent
        Title="{x:Static core:Resources.Strings.AppResources.Home}"
        ContentTemplate="{DataTemplate views:IntroPage}"
        Route="IntroPage">
        <ShellContent.Icon>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE80F;" />
        </ShellContent.Icon>
    </ShellContent>

    <MenuItem
        Text="{x:Static core:Resources.Strings.AppResources.Hemocytometer}"
        BindingContext="{x:Reference root}"
        Clicked="Hemocytometer_Clicked">
        <MenuItem.IconImageSource>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE8EF;" />
        </MenuItem.IconImageSource>
    </MenuItem>

    <MenuItem
        Text="{x:Static core:Resources.Strings.AppResources.BloodDifferential}"
        BindingContext="{x:Reference root}"
        Clicked="BloodDifferential_Clicked">
        <MenuItem.IconImageSource>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE8EF;" />
        </MenuItem.IconImageSource>
    </MenuItem>
    
    <ShellContent
        Title="{x:Static core:Resources.Strings.AppResources.Results}"
        ContentTemplate="{DataTemplate views:ResultsPage}">
        <ShellContent.Icon>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE8BC;" />
        </ShellContent.Icon>
    </ShellContent>

    <MenuItem
        Text="{x:Static core:Resources.Strings.AppResources.Feedback}"
        BindingContext="{x:Reference root}"
        Command="{Binding SendFeedbackCommand}">
        <MenuItem.IconImageSource>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE939;" />
        </MenuItem.IconImageSource>
    </MenuItem>

    <ShellContent
        Title="{x:Static core:Resources.Strings.AppResources.Settings}"
        BindingContext="{x:Reference root}"
        ContentTemplate="{DataTemplate views:SettingsPage}">
        <ShellContent.Icon>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE713;" />
        </ShellContent.Icon>
    </ShellContent>

    <MenuItem
        Text="{x:Static core:Resources.Strings.AppResources.MoreInformation}"
        BindingContext="{x:Reference root}"
        Command="{Binding MoreInformationCommand}">
        <MenuItem.IconImageSource>
            <FontImageSource
                FontFamily="SegoeFluentIcons"
                Color="{AppThemeBinding Light={StaticResource Gray900}, Dark={StaticResource DarkThemeLightTextColor}}"
                Glyph="&#xE946;" />
        </MenuItem.IconImageSource>
    </MenuItem>

</Shell>
```

Comment (BioTurboNick) [reply src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellFlyoutTemplatedContentRenderer.cs:792]: The first two MenuItems execute a Command that uses Navigation.PushAsync to load a new page into the Shell.

**Review by PureWeen — APPROVED** [review]
I realize this issue doesn't have a UITest but none of the referenced issues were able to reproduce this locally and this change is fairly harmless. It's hard to say if it'll manifest in a different way but this might just be a left over measure pass when swapping between different pages.

**Review by mattleibow — APPROVED** [review]

**Pull request merged**